File: easy/1528_shuffle_string/solution.py
from typing import List


class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        shuffled_s = [""] * len(s)

        for i, c in zip(indices, s):
            shuffled_s[i] = c

        return "".join(shuffled_s)

File: easy/1528_shuffle_string/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  std::string restoreString(std::string s, std::vector<int> &indices) {
    std::string shuffledS(s.size(), ' ');

    for (size_t i = 0; i < s.size(); ++i)
      shuffledS[indices[i]] = s[i];

    return shuffledS;
  }
};
